pkg/alertmanager/statefulset.go

```diff
@@ -669,6 +670,30 @@ func makeStatefulSetSpec(logger log.Logger, a *monitoringv1.Alertmanager, config
 		}
 	}
 
+	// Mount web config and web TLS credentials as volumes.
+	// We always mount the web config file for versions greater than 0.22.0.
+	// With this we avoid redeploying alertmanager when reconfiguring between
+	// HTTP and HTTPS and vice-versa.
```

Comment:
I don't think that the comment applies here because an empty mTLS config file isn't allowed. So we'll have to redeploy the pods if the spec changes from no-TLS to TLS (and vice-versa).

Reply:
Right, so we cannot mount an empty volume like we do with web tls. So here, we only mount the config and redeploy the pod depending on whether there is mTLS config or not.

Reply:
// If the am version >= 0.24.0, create a new secret for the mTLS configuration.
	// The secret volume is always created and mounted, even if the secret is empty.
	// But, empty volume cannot be passed as the path in the argument for the cluster TLS configuration.
	// So, we only pass the argument `--cluster.tls-config` if the ClusterTLSConfig field is not nil.